### ISSUE TITLE
Add Version to Template and Pipelines

### ISSUE THREAD

**Issue by pickypg** [elastic/elasticsearch#20171]
When writing code that needs to publish its own template(s) or pipeline(s), it can be a small burden to verify that the existing template version exists, especially if you ever make backward-compatible changes.

I propose that we add a completely optional `version` field to each template and pipeline that is 100% user-driven. If they don't supply it, then it doesn't exist. If they do supply it, then it exists as whatever value they set (regardless of what might have been there before!). The version will exist solely as a mechanism for client software to ensure that not only is their template or pipeline there, but it's the one that they submitted last time without having to go field-by-field.

This allows requests like this to be submitted:

```http
GET /_template/.monitoring-es-2?filter_path=*.version
```

Then, the client can loop through the returned templates and only concern itself with the `name` and `version`. Pipelines would have an equivalent look:

```http
GET /_ingest/pipelines/.monitoring*?filter_path=*.version
```

- [x] Templates #20353
- [x] Pipelines #20343

**Comment by martijnvg** [comment]
I removed the discuss label here. Since the proposed change here makes sense to me.

**Comment by pickypg** [comment]
merged in #20343 and #20353
